model/train.py

```diff
@@ -309,26 +314,68 @@ def train(args):
 
         print("")
         sys.stdout.flush()
-        
+
         if not os.path.exists(args.model_save_dir):
             os.makedirs(args.model_save_dir)
 
-        if args.optimizer == "noam":
-            save_checkpoint({
-                'epoch': epoch,
-                'state_dict': model.state_dict(),
-                'optimizer': optimizer._optimizer.state_dict(),
-            }, "{}/epoch_{}.pth.tar".format(args.model_save_dir, epoch))
-        else:
-            save_checkpoint({
-                'epoch': epoch,
-                'state_dict': model.state_dict(),
-            }, "{}/epoch_{}.pth.tar".format(args.model_save_dir, epoch))
+        if counter < args.num_saved_model:
```

Comment:
There are a lot of duplicated things in both "if" and "else", consider using a function to warp them.